src/components/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { forwardRef, useContext } from 'react';
+import PropTypes from 'prop-types';
+import usePortal from 'react-cool-portal';
+import styled from 'styled-components';
+import { isNotUndefined } from 'ramda-adjunct';
+
+import { DrawerProps } from './Drawer.types';
+import { DrawerSizes } from './Drawer.enums';
+import { useLockBodyScroll } from '../../hooks/useLockBodyScroll';
+import { useOuterClick } from '../../hooks/useOuterCallback';
+import { Inline, Padbox } from '../layout';
+import { H4 } from '../typography';
+import { getColor, getDepth, pxToRem } from '../../utils';
+import { mergeRefs } from '../../utils/mergeRefs';
+import { SpaceSizes } from '../../theme';
+import { DSContext } from '../../theme/DSProvider/DSProvider';
+import { CloseButton } from '../CloseButton';
+import { StretchEnum } from '../layout/Inline/Inline.enums';
+
+const widthVariants = {
+  [DrawerSizes.sm]: 360,
+  [DrawerSizes.md]: 480,
+  [DrawerSizes.lg]: 720,
+  [DrawerSizes.xl]: 960,
+};
+
+const Overlay = styled.div`
+  position: fixed;
+  width: 100%;
+  height: 100%;
+  top: 0;
+  left: 0;
+  background: rgba(0, 0, 0, 0.25);
+  display: flex;
+  align-items: flex-start;
+  justify-content: flex-end;
+  z-index: ${getDepth('modal')};
+`;
+
+const NoOverlay = styled.div`
```

Comment:
This component should not exist, we need interactive body when when overlay is disabled

Reply:
overlay only takes the width of the modal, so it doesn't go over the rest of the page and holds z-index property

Reply:
renamed to Drawer wrapper